services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/MultipartManagerV2Impl.java

```diff
@@ -151,10 +140,25 @@ <T extends MultipartRequest> MultipartUploadStatus startOrResumeMultipartRequest
 		return status.getMultipartUploadStatus();
 	}
 
-	private boolean fileHandleExistsFileNoLongerInCloudStorage(UserInfo user, CompositeMultipartUploadStatus status) {
-		// If the existing upload request is complete, we check if the underlying object has been deleted
-		return (status != null && MultipartUploadState.COMPLETED.equals(status.getMultipartUploadStatus().getState())
-				&& !cloudDaoProvider.getCloudServiceMultipartUploadDao(status.getUploadType()).doesObjectExist(status.getBucket(), status.getKey()));
+	Optional<CompositeMultipartUploadStatus> getCachedUploadOrRestart(UserInfo user, MultipartRequest request, boolean forceRestart) {

```

Comment:
Instead of having the forceRestart in this function and resetting the hash here, the caller can unconditionally reset the hash when a new upload is created. The forceRestart option will make it so that we do not even check for a cached upload. The goal of this method is to only retrieve a potentially cached upload if any that is still valid exist.